File: src/main/java/io/bindernews/thegrackle/helper/HitCountEvent.java
package io.bindernews.thegrackle.helper;

import com.megacrit.cardcrawl.cards.AbstractCard;
import com.megacrit.cardcrawl.core.AbstractCreature;
import lombok.Data;

import java.util.Set;

@Data
public class HitCountEvent {
    final AbstractCreature source;

    /**
     * The card that created this event.
     */
    final AbstractCard card;

    /**
     * Tags indicating damage type, source mod, etc.
     */
    final Set<String> tags;
    final int baseCount;
    int count;

    public void addCount(int deltaCount) {
        count += deltaCount;
    }
}

File: src/main/kotlin/io/bindernews/thegrackle/api/HitCountEvent.kt
package io.bindernews.thegrackle.api

import com.megacrit.cardcrawl.cards.AbstractCard
import com.megacrit.cardcrawl.core.AbstractCreature

data class HitCountEvent(
    /** Creature that originated the event.  */
    val source: AbstractCreature,
    /** The card that created this event.  */
    val card: AbstractCard?,
    /** Tags indicating damage type, source mod, etc.  */
    val tags: Set<String>,
    /** Initial hit count */
    val baseCount: Int = 0,
    /** Mutable hit count */
    var count: Int = 0,
) {
    fun addCount(deltaCount: Int) {
        count += deltaCount
    }
}
